controllers/account/controllers/debt_controller.go

```diff
@@ -0,0 +1,252 @@
+		Updataflag = true
+		if err := r.sendMediumNotice(ctx, account.Name); err != nil {
+			return err
+		}
+		if err := r.change2Medium(ctx, *account); err != nil {
+			return err
+		}
+	}
+
+	mediumtime, err := strconv.Atoi(DebtConfig[string(accountv1.DebtStatusMedium)])
+	if err != nil {
+		r.Error(err, "get small time error")
+	}
+	if debt.Status.AccountDebtStatus == accountv1.DebtStatusMedium && (time.Now().Unix()-debt.Status.LastUpdateTimeStamp) > int64(mediumtime) {
+		debt.Status.AccountDebtStatus = accountv1.DebtStatusLarge
+		debt.Status.LastUpdateTimeStamp = time.Now().Unix()
+		Updataflag = true
+		if err := r.sendLargeNotice(ctx, account.Name); err != nil {
+			return err
+		}
+		if err := r.change2Large(ctx, *account); err != nil {
+			return err
+		}
+	}
+
+	if Updataflag {
+		if err := r.Status().Update(ctx, debt); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *DebtReconciler) change2Normal(ctx context.Context, account accountv1.Account) error {
+	return nil
+}
+
+func (r *DebtReconciler) change2Small(ctx context.Context, account accountv1.Account) error {
+	return nil
+}
+
+func (r *DebtReconciler) change2Medium(ctx context.Context, account accountv1.Account) error {
+	return nil
+}
+
+func (r *DebtReconciler) change2Large(ctx context.Context, account accountv1.Account) error {
+	if err := r.deleteUserResource(ctx, GetUserNameSpace(account.Name)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *DebtReconciler) syncDebt(ctx context.Context, account *accountv1.Account, debt *accountv1.Debt) error {
+	debt.Name = GetDebtName(account.Name)
+	debt.Namespace = "ns-" + account.Name
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, debt, func() error {
+		debt.Spec.UserName = account.Name
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetDebtName(AccountName string) string {
+	return fmt.Sprintf("%s%s", accountv1.DebtPrefix, AccountName)
+}
+
+func GetUserNameSpace(AccountName string) string {
+	return "ns-" + AccountName
+}
+
+func (r *DebtReconciler) sendNotice(ctx context.Context, accountName string, notice any) error {
+	return nil
+}
+
+func (r *DebtReconciler) sendSmallNotice(ctx context.Context, accountName string) error {
+	return r.sendNotice(ctx, accountName, nil)
+}
+
+func (r *DebtReconciler) sendMediumNotice(ctx context.Context, accountName string) error {
+	return r.sendNotice(ctx, accountName, nil)
+}
+func (r *DebtReconciler) sendLargeNotice(ctx context.Context, accountName string) error {
+	return r.sendNotice(ctx, accountName, nil)
+}
+
+func (r *DebtReconciler) deleteUserResource(ctx context.Context, namespace string) error {
+	// delete all deployment
+	deploylist := appsv1.DeploymentList{}
+	if err := r.List(ctx, &deploylist, client.InNamespace(namespace)); client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	for _, deploy := range deploylist.Items {
+		if err := r.Delete(ctx, &deploy); err != nil {
+			return err
+		}
+	}
+
```

Comment:
Do we need delete all crds as well? How about ReplicaSets? DaemonSets?

Reply:
done, add it.